internal/extension/manager.go

```diff
@@ -141,19 +145,44 @@ func newExtensionService(dag *nilGuardedPointer[StateAwareDAG]) extpb.ExtensionS
 }
 
 func (s *extensionService) Subscribe(_ *emptypb.Empty, stream grpc.ServerStreamingServer[extpb.Event]) error {
+	channel := BlockingDAG.newUpdateChannel()
 	for {
-		time.Sleep(time.Second * 5)
-		if err := stream.Send(&extpb.Event{}); err != nil {
-			return err
+		dag := <-channel
+		opts := []cel.EnvOption{
+			kuadrant.CelExt(&dag),
 		}
+
+		s.mutex.Lock()
+		if env, err := cel.NewEnv(opts...); err == nil {
```

Comment:
if there is an error in these cases why not return it? Is the err not of value?

Reply:
I don't want to answer for Alex, but we were discussing this on a call last week - for all of these error cases it would be somewhat useful to pass the error message back to the client instead of just logging it and returning locally.

Reply:
fwiw we agreed I'd handle the error sending in a follow up